From: Andreas Pehrson <apehrson@mozilla.com>
Date: Mon, 27 Feb 2023 16:22:00 +0000
Subject: Bug 1817024 - (fix-fdcfefa708) In PhysicalSocket avoid a non-trivial
 designated initializer. r=mjf,webrtc-reviewers

This fixes a build failure in the base-toolchain job with GCC 7.5.0:
  In file included from Unified_cpp_threading_gn0.cpp:38:0:
  .../third_party/libwebrtc/rtc_base/physical_socket_server.cc: In member function 'int rtc::PhysicalSocket::DoReadFromSocket(void*, size_t, rtc::SocketAddress*, int64_t*)':
  .../third_party/libwebrtc/rtc_base/physical_socket_server.cc:463:51: sorry, unimplemented: non-trivial designated initializers not supported
       msghdr msg = {.msg_iov = &iov, .msg_iovlen = 1};
                                                   ^

Differential Revision: https://phabricator.services.mozilla.com/D171057
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/a3447f709befd84a282ca40f29b7a5ea76d5b68d
---
 rtc_base/physical_socket_server.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rtc_base/physical_socket_server.cc b/rtc_base/physical_socket_server.cc
index 899c29aecd..8c2a96198c 100644
--- a/rtc_base/physical_socket_server.cc
+++ b/rtc_base/physical_socket_server.cc
@@ -528,7 +528,7 @@ int PhysicalSocket::DoReadFromSocket(void* buffer,
   int received = 0;
   if (read_scm_timestamp_experiment_) {
     iovec iov = {.iov_base = buffer, .iov_len = length};
-    msghdr msg = {.msg_iov = &iov, .msg_iovlen = 1};
+    msghdr msg = {.msg_name = nullptr, .msg_namelen = 0, .msg_iov = &iov, .msg_iovlen = 1};
     if (out_addr) {
       out_addr->Clear();
       msg.msg_name = addr;
